File: 20_ValidParentheses/c#/ValidParentheses.cs
public class Solution {
    public bool IsValid(string s) {
        Stack<char> meta = new Stack<char>();
        for(int i = 0; i<s.Length; i++){
            
            if(s[i] == ')'){
                if((meta.Count>0)&&(meta.Peek()=='(')){
                    meta.Pop();
                }
                else{
                    return false;
                }
            }
            else if(s[i] == ']'){
                if((meta.Count>0)&&(meta.Peek()=='[')){
                    meta.Pop();
                }
                else{
                    return false;
                }
            }
            else if(s[i] == '}'){
                if((meta.Count>0)&&(meta.Peek()=='{')){
                    meta.Pop();
                }
                else{
                    return false;
                }
            }
            else{
                meta.Push(s[i]);
            // Console.WriteLine(meta.Peek());
            }
        }
        
        if(meta.Count==0){
            return true;
        }
        else{
            return false;
        }
    }
}

File: 20_ValidParentheses/python/ValidParentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        if len(s) == 0:
            return True
        if len(s)<2:
            return False
        meta = []
        for item in s:
            if item == ')':
                if len(meta)>0 and meta[-1] == '(':
                    meta.pop()
                else:
                    return False
            elif item == ']':
                if len(meta)>0 and meta[-1] == '[':
                    meta.pop()
                else:
                    return False
            elif item == '}':
                if len(meta)>0 and meta[-1] == '{':
                    meta.pop()
                else:
                    return False
            else:
                meta.append(item)
        
        if len(meta) == 0:
            return True
        else:
            return False